### PR TITLE
Use public k8s endpoint by default, disable bastion

### PR DESCRIPTION
**Pull request by robinsturmeit**
This changes the default access pattern from a setup where resources are
private by default and can only be accessed via a bastion host to an
all-public approach. It is the user's responsibility to secure access to
their resources.

The documentation has been updated to reflect this.

Tests:
[x] Checked if clusters exhibit expected access restrictions on GKE, EKS
[x] Checked if modifying endpoint access in EKS is non-destructive

### CODE REVIEW

File: gke/terraform.tfvars.template [hunk 1]
```diff
@@ -14,25 +14,26 @@ node_auto_upgrade = true
...
 enable_nat                     = true
-enable_bastion                 = true
+enable_bastion                 = false
 enable_istio                   = false
 enable_intranode_communication = false
 enable_dashboard               = false
-private_k8s_endpoint           = true
-private_vms                    = true
+# WARNING: Modifying `private_k8s_endpoint` after the cluster has been stood up
+# will DESTROY AND RECREATE the cluster, causing LOSS OF DATA
```
Comment (nsmith5) [inline gke/terraform.tfvars.template:22]: 💯 🥇

File: gke/variables.tf [hunk 2]
```diff
@@ -105,14 +105,14 @@ variable "enable_dashboard" {
 
 variable "private_k8s_endpoint" {
   type        = bool
-  default     = true
-  description = "By default, the Kubernetes endpoint is only accessible via the bastion host. Set to false if you want access via the public internet. You can use IP whitelisting using `allowed_cidr_blocks` to tighten access for both cases."
+  default     = false
+  description = "Setting this to true will disable access to the k8s API via the public internet. You will need a bastion or VPN to operate the k8s cluster"
```
Comment (nsmith5) [inline gke/variables.tf:109]: This is fairly different than EKS (only one variable, only for private and default false), is it worth making the options similar on each platform? Or will that introduce breaking changes?

Comment (robinsturmeit) [reply gke/variables.tf:109]: That's just something that's not supported by Terraform for GKE. Either enable private endpoint and disable public endpoint or use public endpoint.

**Review by lloydChris — APPROVED** [review]
small nit/opinion.  But otherwise LGTM

File: gke/README.md [hunk 3]
```diff
@@ -153,6 +143,9 @@ The leading `--` says "pass the following arguments down to the underlying
 forward creds to the next machine for you so that you don't have to copy a key
 around.
 
+Alternatively, you may set up a VPN connection into your VPC to access resources
+using their internal IPs.
+
```
[the remark this answers is not in the document]
Comment (robinsturmeit) [reply gke/README.md:148]: By the same token we could remove the entire section. Hmmm... Maybe we should. I suggest I add this as an item to the ticket dealing with removing the bastion completely. Agree?

File: gke/variables.tf [hunk 4]
```diff
@@ -75,7 +75,7 @@ variable "enable_nat" {
 
 variable "enable_bastion" {
   type        = bool
-  default     = true
+  default     = false
   description = "Include a bastion/jump server in deployment. You can restrict the range of IPs that can connect to the bastion using `allowed_cidr_blocks`"
```
Comment (yugesh84) [inline gke/variables.tf:79]: Worth calling out that this is deprecated and will be removed shortly?

**Review by yugesh84 — APPROVED** [review]
One small comment, and the failing `validate` job - otherwise LGTM

File: gke/variables.tf (hunk 4 above)
Comment (robinsturmeit) [reply gke/variables.tf:79]: Wouldn't the release notes be the appropriate place for this? Or you think we should also do it here? Or in there `terraform.tfvars.template` file? All three?

Comment (yugesh84) [reply gke/variables.tf:79]: Hmm, along with release notes, the template file seems to make the most sense to me - as that is probably what a new customer would see.

Comment (robinsturmeit) [reply gke/variables.tf:79]: Sounds good. Fixed the failing `fmt` check and added a note on the deprecation of the bastion host setting

File: gke/variables.tf (hunk 2 above)
Comment (nsmith5) [reply gke/variables.tf:109]: Ah! I see! Thanks

**Pull request merged**